### PR TITLE
feat: populate Programmed condition for KongConsumerGroup

### PR DESCRIPTION
**Pull request by programmer04**
**What this PR does / why we need it**:

Populates Programmed condition for `KongConsumerGroup` and prints an additional column `PROGRAMMED` for this CRD as introduced in https://github.com/Kong/kubernetes-ingress-controller/pull/4425 for others. 

**Which issue this PR fixes**:

Part of https://github.com/Kong/kubernetes-ingress-controller/issues/3728

**Special notes for your reviewer**:

Something to discuss - I decided to check in the integration test if the condition is populated correctly because the duration of this check is ~3 seconds, hence it's negligible for an overall length of execution ~2.5%. Unfortunately, it has been written in this verbose way, because https://github.com/stretchr/testify/issues/1391 hasn't been resolved yet.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: test/integration/consumer_group_test.go [hunk 1]
```diff
@@ -86,4 +88,35 @@ func TestConsumerGroup(t *testing.T) {
...
+		if status.Status != metav1.ConditionTrue {
+			t.Logf("WARNING: expected status: %q, got: %q", metav1.ConditionTrue, status.Status)
+			return false
+		}
+		if status.Type != string(kongv1.ConditionProgrammed) {
+			t.Logf("WARNING: expected type: %q, got: %q", kongv1.ConditionProgrammed, status.Type)
+			return false
+		}
+		return true
+	}, ingressWait, waitTick)
```
Comment (czeslavo) [inline test/integration/consumer_group_test.go:121]: Because we have an envtest covering this already, I'm on the fence about whether we need to duplicate this in integration tests as well. 🤔

Comment (programmer04) [reply test/integration/consumer_group_test.go:121]: Yup, I was wondering too. I added it because it's pretty cheap to have and easy to add (especially firstly when `EventuallyWithT()` is used). I thought about testing it when the consumer, plugin, and consumer group will be configured. But as well I can remove it as, as you said it's already in envtests. Do you have a strong preference?

Comment (czeslavo) [reply test/integration/consumer_group_test.go:121]: Well, in my opinion, it's better to keep this kind of assertion in envtests as they're not dependent on the integration with Kong. envtest already covers the `Programmed` condition E2E. Adding the assertion to the integration test in my feeling is unnecessary duplication of this. 

I'd be interested in @pmalek opinion on that. :)

Comment (pmalek) [reply test/integration/consumer_group_test.go:121]: 👍 I don't see a reason either to test it if it's already covered with `envtest` based test.

The only point I'd like to make it so be sure we cover the corner cases w.r.t API differences between dbless and db backed in Admin API mocks (and ideally mention that explicitly). That was a surprise to me when I saw a special case for db backed and dbless recently in one of the PRs for Consumer Groups

Comment (programmer04) [reply test/integration/consumer_group_test.go]: I got rid of it

**Comment by czeslavo** [comment]
Can you please add `KongConsumerGroup` to the list of CRDs extended with `Programmed` condition in the last two [changelog](https://github.com/Kong/kubernetes-ingress-controller/blob/964f9d264246955a11e4a8bbe778e84c66b09a3c/CHANGELOG.md?plain=1#L130C10-L130C10) entries?

**Comment by programmer04** [comment]
Good point @czeslavo, I've enhanced those entries

**Review by czeslavo — APPROVED** [review]
